File: Artist/app/src/main/java/com/example/artist/ui/toptrackslisting/di/TopTracksComponent.java
package com.example.artist.ui.toptrackslisting.di;


import com.example.artist.ui.toptrackslisting.TopTracksFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {TopTracksModule.class})
public interface TopTracksComponent {
    void inject(TopTracksFragment target);
}

File: Artist/app/src/main/java/com/example/artist/ui/toptrackslisting/di/TopTracksComponent.kt
package com.example.artist.ui.toptrackslisting.di

import com.example.artist.ui.toptrackslisting.TopTracksFragment
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [TopTracksModule::class])
interface TopTracksComponent {
    fun inject(target: TopTracksFragment)
}